src/status_im/contexts/wallet/sheets/buy_token/view.cljs

```diff
@@ -1,34 +1,64 @@
 (ns status-im.contexts.wallet.sheets.buy-token.view
-  (:require [quo.core :as quo]
+  (:require [oops.core :as oops]
+            [quo.core :as quo]
             [react-native.core :as rn]
             [status-im.contexts.wallet.sheets.buy-token.style :as style]
+            [status-im.feature-flags :as ff]
             [utils.i18n :as i18n]
             [utils.re-frame :as rf]))
 
 (defn- crypto-on-ramp-item
-  [{:keys [name description fees logo-url site-url]}]
-  [quo/settings-item
-   {:title             name
-    :description       :text
-    :description-props {:text description}
-    :tag               :context
-    :tag-props         {:icon    :i/fees
-                        :context fees}
-    :action            :arrow
-    :action-props      {:alignment :flex-start
-                        :icon      :i/external}
-    :image             :icon-avatar
-    :image-props       {:icon logo-url}
-    :on-press          #(rn/open-url site-url)}])
+  [{:keys [name description fees logo-url site-url recurrent-site-url]} _ _ {:keys [tab]}]
+  (let [open-url (rn/use-callback (fn []
+                                    (rn/open-url (if (= tab :recurrent) recurrent-site-url site-url)))
+                                  [site-url recurrent-site-url tab])]
+    [quo/settings-item
+     {:title             name
+      :description       :text
+      :description-props {:text description}
+      :tag               :context
+      :tag-props         {:icon    :i/fees
+                          :context fees}
+      :action            :arrow
+      :action-props      {:alignment :flex-start
+                          :icon      :i/external}
+      :image             :icon-avatar
+      :image-props       {:icon logo-url}
+      :on-press          open-url}]))
+
+(def ^:private tabs
+  [{:id    :one-time
+    :label (i18n/label :t/one-time)}
+   {:id    :recurrent
+    :label (i18n/label :t/recurrent)}])
+
+(def ^:private initial-tab (:id (first tabs)))
 
 (defn view
   []
   (rn/use-mount (fn []
                   (rf/dispatch [:wallet/get-crypto-on-ramps])))
-  (let [crypto-on-ramps (rf/sub [:wallet/crypto-on-ramps])]
+  (let [one-time-crypto-on-ramps        (rf/sub [:wallet/one-time-crypto-on-ramps])
+        recurrent-crypto-on-ramps       (rf/sub [:wallet/recurrent-crypto-on-ramps])
+        [selected-tab set-selected-tab] (rn/use-state initial-tab)
+        [min-height set-min-height]     (rn/use-state 0)
+        on-layout                       (rn/use-callback
+                                         #(set-min-height
+                                           (oops/oget % :nativeEvent :layout :height)))]
     [:<>
      [quo/drawer-top {:title (i18n/label :t/buy-assets)}]
+     (when (ff/enabled? ::ff/wallet.buy-recurrent-assets)
```

Comment:
since there's a lot of changes in the component and to avoid the layout calculations (if i understood correctly it's to maintain the layout of the component as it was without), wouldn't it be better to just have two separate components that are rendered depending on the feature flag?

Reply:
This is a temporary feature flag that we will remove once this backend task has been done: https://github.com/status-im/status-desktop/issues/14818. 

Right now, even though we have all the logic in place, we don't have any providers that support recurrent purchases. That's why we are hiding that part with a flag.